models/ContainerPage.php

```diff
@@ -67,10 +69,84 @@ public function rules()
     {
         $rules = $this->defaultRules();
         $rules[] = ['in_new_window', 'integer'];
-        $rules[] = [['page_content'], 'safe'];
+	    $rules[] = [['page_content'], 'safe', 'when' => array($this, 'isNotHtml')];
+	    $rules[] = [['page_content'], 'filter', 'when' => array($this, 'isHtml'), 'filter' => array($this, 'purifyFilter')];
+
         return $rules;
     }
 
+	/**
+	 * @param ContainerPage $model the HTML code container
+	 * @return bool whether it is an html type container
+	 */
+	public function isHtml($model)
+	{
+		return $model->type == Container::TYPE_HTML;
+	}
+
+	/**
+	 * @param ContainerPage $model the HTML code
+	 * @return bool whether it is not an html type container
+	 */
+	public function isNotHtml($model)
+	{
+		return $model->type != Container::TYPE_HTML;
+	}
+
+	/**
+	 * Purify the HTML code only if its container type is html (conditional validation)
+	 *
+	 * @param string $html the HTML code to be purified
+	 * @return string the purified HTML code
+	 */
+    public function purifyFilter($html)
+    {
+	    return HtmlPurifier::process($html, $this->purifierConfig('test'));
+    }
+
+	/**
+	 * Returns HTMLPurifier configuration set.
+	 *
+	 * @param string $type set name
+	 * @return array configuration
+	 */
+    private function purifierConfig($type = '')
+    {
+	    // TODO: make workable safe config presets
+	    // test preset is only for testing
+	    // other presets will be discussed
```

Comment:
I think for now we can just always use the default or full preset? to me that whitelist sounds good - provides enough flexibility.

Reply:
I was thinking maybe add an option in the module settings, to select one of the preset, another proposition was to add the possibility to change directly the whitelist from within the module settings as a text field?

Reply:
I see. good idea! 
but I'd keep it simple for now and try to do the most simple thing that works.